compiler/rustc_errors/src/emitter.rs

```diff
@@ -2559,60 +2559,66 @@ fn num_decimal_digits(num: usize) -> usize {
 
 // We replace some characters so the CLI output is always consistent and underlines aligned.
 // Keep the following list in sync with `rustc_span::char_width`.
+// ATTENTION: keep lexicografically sorted so that the binary search will work
```

Comment:
is this worth adding a `// tidy-sort-....` or a doctest or static assert to enforce? (if it is possible to enforce)

Reply:
Done

Reply:
Maybe a test (or a const assert?) should be added too? I mean, `is_sorted` got stabilized just recently #128279, and some things might slip past tidy, considering how `\r`, `\u{0d}`, and `\u{000d}` mean the same but get sorted differently.

Reply:
Another idea: maybe use perfect hashing for this? Will probably have to benchmark it.

Reply:
@GrigorenkoPV that's a reasonable idea. I am not *too* concerned with this because it is a part of the codebase that changes *really* rarely. The point about the same byte sequence being represented in different textual representations is an issue even without the sorting: what if you have all three representations of `\r` in the list with different replacements? That's another bad failure mode.

Reply:
> what if you have all three representations of `\r` in the list with different replacements?

That's exactly what reminded me of perfect hashing. In any case, none of this is in scope of this PR and can be tackled later, separately.